packages/plugin-ext/src/main/browser/notebooks/notebookDto.ts

```diff
@@ -0,0 +1,135 @@
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+
+import * as notebookCommon from '@theia/notebook/lib/common';
+import * as rpc from '../../../common';
+
+export namespace NotebookDto {
+
+    export function toNotebookOutputItemDto(item: notebookCommon.OutputItemDto): rpc.NotebookOutputItemDto {
+        return {
+            mime: item.mime,
+            valueBytes: item.data
+        };
+    }
+
+    export function toNotebookOutputDto(output: notebookCommon.OutputDto): rpc.NotebookOutputDto {
+        return {
+            metadata: output.metadata,
+            items: output.outputs.map(toNotebookOutputItemDto)
+        };
+    }
+
+    export function toNotebookCellDataDto(cell: notebookCommon.CellDto): rpc.NotebookCellDataDto {
+        return {
+            cellKind: cell.cellKind,
+            language: cell.language,
+            source: cell.source,
+            internalMetadata: cell.internalMetadata,
+            metadata: cell.metadata,
+            outputs: cell.outputs.map(toNotebookOutputDto)
+        };
+    }
+
+    export function toNotebookDataDto(data: notebookCommon.NotebookData): rpc.NotebookDataDto {
+        return {
+            metadata: data.metadata,
+            cells: data.cells.map(toNotebookCellDataDto)
+        };
+    }
+
+    export function fromNotebookOutputItemDto(item: rpc.NotebookOutputItemDto): notebookCommon.OutputItemDto {
+        return {
+            mime: item.mime,
+            data: item.valueBytes
+        };
+    }
+
+    export function fromNotebookOutputDto(output: rpc.NotebookOutputDto): notebookCommon.OutputDto {
+        return {
+            metadata: output.metadata,
+            outputs: output.items.map(fromNotebookOutputItemDto)
+        };
+    }
+
+    export function fromNotebookCellDataDto(cell: rpc.NotebookCellDataDto): notebookCommon.CellDto {
+        return {
+            cellKind: cell.cellKind,
+            language: cell.language,
+            source: cell.source,
+            outputs: cell.outputs.map(fromNotebookOutputDto),
+            metadata: cell.metadata,
+            internalMetadata: cell.internalMetadata
+        };
+    }
+
+    export function fromNotebookDataDto(data: rpc.NotebookDataDto): notebookCommon.NotebookData {
+        return {
+            metadata: data.metadata,
+            cells: data.cells.map(fromNotebookCellDataDto)
+        };
+    }
+
+    export function toNotebookCellDto(cell: notebookCommon.Cell): rpc.NotebookCellDto {
+        return {
+            handle: cell.handle,
+            uri: cell.uri.toComponents(),
+            source: cell.textBuffer.split('\n'),
+            eol: '\n',
```

Comment:
Does this work as expected on Windows? Maybe something like this is more correct:

```suggestion
        const eol = OS.backend.isWindows ? '\r\n' : '\n';
        return {
            handle: cell.handle,
            uri: cell.uri.toComponents(),
            source: cell.textBuffer.split(eol),
            eol,
```